examples/object_detection_3d/object_detection_3d/evaluator.py

```diff
@@ -77,11 +86,53 @@ def name(self):
         return name
 
 
+@dataclass(frozen=True)
+class UnmatchedLabeledBoundingBox3D(LabeledBoundingBox3D):
+    max_overlap: float
+
+    @staticmethod
+    def from_bbox(bbox: LabeledBoundingBox3D, overlap: float) -> "UnmatchedLabeledBoundingBox3D":
+        return UnmatchedLabeledBoundingBox3D(
+            label=bbox.label,
+            center=bbox.center,
+            dimensions=bbox.dimensions,
+            rotations=bbox.rotations,
+            max_overlap=overlap,
+        )
+
+
+@dataclass(frozen=True)
+class UnmatchedScoredBoundingBox3D(ScoredLabeledBoundingBox3D):
+    max_overlap: float
+
+    @staticmethod
+    def from_bbox(bbox: ScoredLabeledBoundingBox3D, overlap: float) -> "UnmatchedScoredBoundingBox3D":
+        return UnmatchedScoredBoundingBox3D(
+            label=bbox.label,
+            score=bbox.score,
+            center=bbox.center,
+            dimensions=bbox.dimensions,
+            rotations=bbox.rotations,
+            max_overlap=overlap,
+        )
+
+
 @dataclass(frozen=True)
 class MetricsTestSample(BaseMetricsTestSample):
-    nObjects: int
     nInferences: int
     nValidObjects: int
+    thresholds: List[ScoredClassificationLabel]
+    nMatchedInferences: int
+    nMissedObjects: int
+    nMismatchedInferences: int
+    valid_2D: List[LabeledBoundingBox]
```

Comment:
This is the replacement for previous `n_easy`, `n_moderate`, and `n_hard`, where only ground_truths matching the target EvaluatorConfiguration (easy, moderate, or hard) are considered valid.